### ISSUE TITLE
DTensor linear fails for ConvRot INT8 and silently miscomputes ConvRot NVFP4

### ISSUE THREAD

**Issue by Boffee** [Boffee/piper-kernels#106]
Follow-up to #104, tested after the view fix in `4449e9db6965a608369041af7e8b30d1e11bc00a`. `DTensor.from_local(...).to_local()` now preserves all three wrappers, and all 63 view tests pass. Calling ordinary `torch.nn.functional.linear` on these DTensors exposes the next problem: ConvRot INT8 cannot transpose its weight, while the two NVFP4 wrappers become plain TorchAO `NVFP4Tensor` during transpose. ConvRot NVFP4 then returns incorrect results because its rotation semantics are lost.

This reproduces with one standard Gloo rank and replicated CUDA DTensors. No Piper Offload executor, custom process group, model-specific code, or data collective is involved.

### Observed behavior

| Weight wrapper | `dw.t().to_local()` | `F.linear(dx, dw)` |
| --- | --- | --- |
| `ConvRotInt8Tensor` | Raises `NotImplementedError` for `aten.t.default` | Raises the same error |
| `PiperNVFP4Tensor` | Plain TorchAO `NVFP4Tensor`; `high_first` missing | Runs after losing the Piper wrapper/dispatch |
| `ConvRotNVFP4Tensor` | Plain TorchAO `NVFP4Tensor`; `group_size` and `high_first` missing | Runs with incorrect output |

The reference is `F.linear(x, weight)` using the **same already-quantized local Piper weight**, not the original BF16 weight. In the single-rank reproduction below, ConvRot NVFP4 has relative L2 error approximately **1.3033 (130%)** and maximum absolute error **1.6602**. Piper NVFP4 has a much smaller discrepancy (relative L2 approximately 0.0112); that discrepancy alone is not a correctness verdict, but the wrapper loss changes its dispatch path.

Separate two-rank validation reproduced the eager failures for `Replicate()`, `Shard(0)` and `Shard(1)`. Replicated fullgraph `torch.compile` validation with each format's Piper compile options and CUDA graphs disabled also fails for INT8 and produces incorrect ConvRot NVFP4 output (relative L2 approximately 1.3224). Compilation therefore does not resolve this.

### Reproduction

Environment: RTX 5090, Linux, Python 3.14, PyTorch `2.13.0+cu130`, TorchAO `0.18.0+cu130`, Kernels source at the commit above. Run with the Kernels checkout's `src` on `PYTHONPATH` and its dependencies installed. This checks eager execution and does not require Piper Offload.

```python
import torch
import torch.distributed as dist
import torch.nn.functional as F
from torch.distributed.device_mesh import DeviceMesh
from torch.distributed.tensor import DTensor, Replicate
from torchao.prototype.mx_formats.nvfp4_tensor import QuantizeTensorToNVFP4Kwargs

from piper_kernels.linear.convrot.int8 import ConvRotInt8Tensor
from piper_kernels.linear.convrot.nvfp4 import ConvRotNVFP4Tensor
from piper_kernels.linear.nvfp4 import PiperNVFP4Tensor

torch.set_num_threads(1)
torch.manual_seed(19)
dist.init_process_group("gloo", store=dist.HashStore(), rank=0, world_size=1)
try:
    mesh = DeviceMesh("cuda", [0])
    with torch.no_grad():
        for cls in (ConvRotInt8Tensor, PiperNVFP4Tensor, ConvRotNVFP4Tensor):
            source = torch.randn(256, 256, device="cuda", dtype=torch.bfloat16) * 0.02
            if cls is ConvRotInt8Tensor:
                weight = cls.from_hp(source, group_size=64)
            else:
                weight = cls.from_hp(
                    source,
                    compute_per_tensor_scale=True,
                    is_swizzled_scales=True,
                    act_quant_kwargs=QuantizeTensorToNVFP4Kwargs(
                        block_size=16,
                        is_swizzled_scales=True,
                        use_triton_kernel=False,
                        use_dynamic_per_tensor_scale=True,
                    ),
                    **({"group_size": 64} if cls is ConvRotNVFP4Tensor else {}),
                )
            x = torch.randn(32, 256, device="cuda", dtype=torch.bfloat16)
            expected = F.linear(x, weight).float()
            dw = DTensor.from_local(weight, mesh, [Replicate()], run_check=False)
            dx = DTensor.from_local(x, mesh, [Replicate()], run_check=False)
            assert type(dw.to_local()) is cls  # The view fix works.
            print(cls.__name__, flush=True)
            try:
                transposed = dw.t().to_local()
                print("  transpose:", type(transposed).__name__,
                      "group_size:", getattr(transposed, "group_size", "MISSING"),
                      "high_first:", getattr(transposed, "high_first", "MISSING"))
            except NotImplementedError as error:
                print("  transpose:", error)
            try:
                actual = F.linear(dx, dw).to_local().float()
                print("  relative L2 error:", ((actual - expected).norm() / expected.norm()).item())
                print("  max absolute error:", (actual - expected).abs().max().item())
            except NotImplementedError as error:
                print("  linear:", error)
finally:
    dist.destroy_process_group()
```

### Expected behavior

Ordinary DTensor linear should preserve the quantized weight's interpretation and produce outputs consistent with the corresponding local Piper linear implementation. The transpose/matmul path used by DTensor needs to retain the Piper dispatch and relevant metadata, including ConvRot rotation parameters and NVFP4 packing order. Adding transpose support alone should not be treated as complete unless the subsequent matmul also uses the correct quantized semantics.

Please cover actual `F.linear` execution in regression tests, including eager and compiled execution, with the local quantized result as a numerical reference. The singleton replicated case above isolates the failure; row/column sharding should also be covered when validating tensor-parallel support. Unsupported paths should raise explicitly instead of returning a base wrapper that silently misinterprets the weight.
